src/frontend/src/instance/distributed.rs

```diff
@@ -140,14 +140,17 @@ impl DistInstance {
         Ok(Output::AffectedRows(0))
     }
 
-    async fn handle_sql(&self, sql: &str, query_ctx: QueryContextRef) -> Result<Output> {
-        let stmt = parse_stmt(sql)?;
+    async fn handle_statement(
+        &self,
+        stmt: Statement,
+        query_ctx: QueryContextRef,
+    ) -> Result<Output> {
         match stmt {
             Statement::Query(_) => {
                 let plan = self
                     .query_engine
                     .statement_to_plan(stmt, query_ctx)
-                    .context(error::ExecuteSqlSnafu { sql })?;
+                    .context(error::ExecuteStatementSnafu {})?;
```

Comment:
Do we need to include `stmt` in error?

Reply:
We cannot because it's taken by `query_engine`. It's query_engine's responsibility to include statement in its error.